### ISSUE TITLE
Asyncio NotImplemented Error with Python 3.8 and Windows

### ISSUE THREAD

**Issue by waylan** [lepture/python-livereload#209]
As explained in tornadoweb/tornado#2608, the default behavior in Python 3.8 has changed and we get an error on Windows only:

```
>livereload site
[I 191026 22:40:22 server:296] Serving on http://127.0.0.1:35729
Traceback (most recent call last):
  File "c:\program files\python38\lib\runpy.py", line 192, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\program files\python38\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Program Files\Python38\Scripts\livereload.exe\__main__.py", line 7, in <module>
  File "c:\program files\python38\lib\site-packages\livereload\cli.py", line 60, in main
    server.serve(host=args.host, port=args.port, root=args.directory,
  File "c:\program files\python38\lib\site-packages\livereload\server.py", line 298, in serve
    self.application(
  File "c:\program files\python38\lib\site-packages\livereload\server.py", line 253, in application
    app.listen(port, address=host)
  File "c:\program files\python38\lib\site-packages\tornado\web.py", line 2112, in listen
    server.listen(port, address)
  File "c:\program files\python38\lib\site-packages\tornado\tcpserver.py", line 152, in listen
    self.add_sockets(sockets)
  File "c:\program files\python38\lib\site-packages\tornado\tcpserver.py", line 165, in add_sockets
    self._handlers[sock.fileno()] = add_accept_handler(
  File "c:\program files\python38\lib\site-packages\tornado\netutil.py", line 279, in add_accept_handler
    io_loop.add_handler(sock, accept_handler, IOLoop.READ)
  File "c:\program files\python38\lib\site-packages\tornado\platform\asyncio.py", line 99, in add_handler
    self.asyncio_loop.add_reader(fd, self._handle_events, fd, IOLoop.READ)
  File "c:\program files\python38\lib\asyncio\events.py", line 501, in add_reader
    raise NotImplementedError
NotImplementedError
```

It appears that Tornado is not going to override the default themselves, but instead expect applications using Tornado to override the default. In other words, Tornado expects Python-Livereload to set this itself.

Of course users of Python-Livereload who are calling the lib programmatically through Python code can just do the override themselves. However, anyone using the command line tool will need Python-Livereload to address this for them. Therefore, my question is: will Python-Livereload be addressing this for all users or only the command line tool?

**Comment by waylan** [comment]
Could we get a response on this? At a minimum, what is the plan? As the developer of a third party tool which calls Python-Livereload through the Python API, should I be addressing this myself, or will you be addressing it?

**Comment by playerla** [comment]
Seems related to a breaking change in tornado : https://github.com/tornadoweb/tornado/issues/2608
For quick reference one random fix : https://github.com/dask/distributed/pull/3249/commits/88373652f7856e3c8df79a142a93b7f4f0786290

**Comment by playerla** [comment]
I test this one in server.py and seems to work
https://github.com/python-telegram-bot/python-telegram-bot/pull/1614/commits/8b4453f4387e19fab0c9da35218dcb2ab7857a58

**Comment by bmcculley** [comment]
A fix for this has just been merged into master.

**Comment by waylan** [comment]
Thanks @bmcculley.

For those who care, the fix was applied in PR #218 and commit e0ea86a. As the fix is included in `server.py`, not `cli.py` it applies to both the python lib and the CLI. No need for anyone using this as a lib to apply the fix themselves. That is, once we get the next release.

**Comment by bmcculley** [comment]
Closing this, new release on pypi fixes this issue.
